runtime/lua/vim/lsp.lua

```diff
@@ -567,10 +567,20 @@ local function buf_attach(bufnr)
     on_detach = function()
       local params = { textDocument = { uri = uri } }
       for _, client in ipairs(lsp.get_clients({ bufnr = bufnr })) do
+        api.nvim_exec_autocmds('LspDetach', {
+          buffer = bufnr,
+          modeline = false,
+          data = { client_id = client.id },
+        })
```

Comment:
Q: what happens if the autocmd handler throws an error?

Reply:
Are errors propagated through nvim_exec_autocmds?

Reply:
Tested with a simple handler that just unconditionally errors, and it doesn't seem to break anything for a normal LSP. I say normal because it was breaking things with null-ls (none-ls), but the error was coming from the `client.notify` call, which null-ls monkey-patches afaik. Not sure if this is something that should be accounted for.